src/IncludedCommands/Persistent/pbans.luau

```diff
@@ -51,86 +50,275 @@ local function GetCachedUsername(UserId: number): string
+                            table.insert(Lines, `User: @{GetCachedUsername(ErrorMessage.UserId)} ({ErrorMessage.UserId})`)
+                        end
+                        for _, Line in ErrorMessage.Lines do
+                            table.insert(Lines, Line)
+                        end
+                    end
+                end
+                TextList:SetLines(Lines)
+            end
+
+            UsernameLoaded.Event:Connect(UpdateText)
+
+            task.spawn(UpdateText)
+            Window:EnableClosing()
+            Window:EnableResizing()
+            Window:Show()
+        end)
     end,
-    ClientRun = function(CommandContext: Types.CmdrCommandContext)
+    ClientRun = function(CommandContext: Types.CmdrCommandContext, Names: {string})
         local Util = IncludedCommandUtil.ForContext(CommandContext)
         local Api = Util:GetClientApi()
 
-        --Show the window.
-        local Window = Api.Window:CreateWindow("Persistent Bans")
-        local TextList = Window:AddTextList()
-        TextList:EnableSearching()
-
-        local function UpdateText()
-            TextList:SetLines({"Loading..."})
-            
-            --Get the bans.
-            local Bans, NewUsernameCache = (Util:GetRemote("GetPersistentBans") :: RemoteFunction):InvokeServer()
-            for UserId, Username in NewUsernameCache do
-                if not UsernameCache[UserId] then
-                    UsernameCache[UserId] = Username
-                end
+        local Bans, NewUserNameCache = (Util:GetRemote("GetPersistentBans") :: RemoteFunction):InvokeServer(Names)
+
+        if typeof(Bans) == "string" then
+            return Bans
+        end
+
+        for UserId, Username in NewUserNameCache do
+            if not UsernameCache[UserId] then
+                UsernameCache[UserId] = Username
             end
+        end
 
-            --Format the lines.
-            --Username fetching is done on the client in case the server didn't load it in time.
-            local Lines = {}
-            if typeof(Bans) == "string" then
-                Lines = {Bans}
-            else
-                local NewLines = {}
-                for UserId, Message in Bans do
-                    if Message == true then
-                        table.insert(NewLines, `{GetCachedUsername(UserId)} ({UserId})`)
-                    else
-                        table.insert(NewLines, `{GetCachedUsername(UserId)} ({UserId}) - {Message}`)
+        for UserId, History: {{BanHistoryPage}} in Bans do
+            --Show the window.
+            local Window = Api.Window:CreateWindow(`@{GetCachedUsername(UserId)} ({UserId})'s Ban History`)
+            local TextList = Window:AddTextList()
+            TextList:EnableSearching()
+
+            local function UpdateText()
+                TextList:SetLines({"Loading..."})
+                Window:SetTitle(`@{GetCachedUsername(UserId)} ({UserId})'s Ban History`)
+
+                local Lines = {}
```

Comment:
There are downstream typing issues with a mix of strings and formatted strings. Either converting to rich text (`"<b>...</b>"`) or declaring the type of `Lines` as `{} :: {string | {Text: string, Font: Enum.Font}}` would fix this.